File: CPP VW/CPP VW/Turtle.cpp
#include "Turtle.h"
#include "World.h"

Turtle::Turtle(World& worldref, int x, int y) :
	Animal(worldref, x, y)
{
	this->power = 2;
	this->initiative = 1;
	this->sign = 'T';
}
Turtle::~Turtle() {}
bool Turtle::specialCollision(Organism* other) {
	if (other->GetPower() >= 5)
		return false;
	else {
		return true;
	}
}
bool Turtle::specialAction() {
	int x = rand() % 4;
	if (x == 0)
		this->Move();
	return true;
}

File: JAVA VW/src/programowanie/obiektowe/proj/Turtle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package programowanie.obiektowe.proj;
import java.util.Random;
/**
 *
 * @author kubaz
 */
public class Turtle extends Animal{

    public Turtle(Position position, World world) {
        super(position, world);
        this.name="Turtle";
        this.initiative=1;
        this.strength=2;
    }
    
    @Override 
    public void action(){
        this.movePermission=true;
        Random random = new Random();
        if(random.nextInt(4)<3)
            return;
        move(world.randomAround(location, 1));
    }
    
      @Override
    public boolean specialCollision(Organism other){
        if(other.getStrength()<5){
            other.movePermission=false;
            return false;
        }  
        return true;    
    }
}
